[quant] Adding Scalar add/mul.

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25080 [quant] Rewriting the scalar add
* #24873 [quant] Rewrite of the scalar mul
* **#24447 [quant] Adding Scalar add/mul.**

Note: This should be landed ONLY after #24259

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24447

Differential Revision: [D16846006](https://our.internmc.facebook.com/intern/diff/D16846006)

Commit:
Adding Scalar add/mul.

Note: This should be landed ONLY after #24259

lly-zero-one reviewed (commented):
LGTM, can you also add a test to cover the add/mul scalar?

llyfacebook commented on aten/src/ATen/native/quantized/cpu/qadd.cpp:
```diff
@@ -48,6 +48,28 @@ Tensor _add_out(Tensor& out, const Tensor& self, const Tensor& other) {
+  AT_DISPATCH_QINT_TYPES(out.scalar_type(), "qadd", [&]() {
+    cpu_kernel(iter, [&](scalar_t a) -> scalar_t {
+      const auto da = at::dequantize_val(self_scale, self_zero_point, a);
+      float c = da + float(other);
```
call the scalar to float conversion function?

Commit:
Update on "Adding Scalar add/mul."


Note: This should be landed ONLY after #24259

Differential Revision: [D16846006](https://our.internmc.facebook.com/intern/diff/D16846006)

Commit:
Update on "Adding Scalar add/mul."


Note: This should be landed ONLY after #24259

Differential Revision: [D16846006](https://our.internmc.facebook.com/intern/diff/D16846006)

zafartahirov commented:
> LGTM, can you also add a test to cover the add/mul scalar?

The numerical tests are performed on the kernels. The API test is covered by the scalability

Commit:
Update on "Adding Scalar add/mul."


Note: This should be landed ONLY after #24259

Differential Revision: [D16846006](https://our.internmc.facebook.com/intern/diff/D16846006)

Commit:
Update on "Adding Scalar add/mul."


Note: This should be landed ONLY after #24259

Differential Revision: [D16846006](https://our.internmc.facebook.com/intern/diff/D16846006)

Commit:
Update on "Adding Scalar add/mul."


Note: This should be landed ONLY after #24259

Differential Revision: [D16846006](https://our.internmc.facebook.com/intern/diff/D16846006)

Commit:
Update on "[quant] Adding Scalar add/mul."


Note: This should be landed ONLY after #24259

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24447

Differential Revision: [D16846006](https://our.internmc.facebook.com/intern/diff/D16846006)

llyfacebook commented on aten/src/ATen/native/quantized/cpu/qadd.cpp:
```diff
@@ -66,30 +88,69 @@ class QAdd final : public c10::OperatorKernel {
 class QAddOut final : public c10::OperatorKernel {
  public:
-  Tensor operator()(at::Tensor qa, at::Tensor qb, at::Tensor out) {
+  Tensor operator()(Tensor qa, Tensor qb, at::Tensor out) {
```
Can you make the namespace be consistent?

lly-zero-one reviewed (approved):
looks good!

Commit:
Update on "[quant] Adding Scalar add/mul."


Note: This should be landed ONLY after #24259

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24447

Differential Revision: [D16846006](https://our.internmc.facebook.com/intern/diff/D16846006)

Commit:
Update on "[quant] Adding Scalar add/mul."


Note: This should be landed ONLY after #24259

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24447

Differential Revision: [D16846006](https://our.internmc.facebook.com/intern/diff/D16846006)
